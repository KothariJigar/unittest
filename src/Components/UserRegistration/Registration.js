import React, { useState } from "react";
import { useNavigate } from "react-router-dom";

export default function Registration() {
  const navigate = useNavigate();
  const [name, setName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [city, setCity] = useState("");
  const [age, setAge] = useState("");
  const [error, setError] = useState("");

  const handleAdd = (e) => {
    e.preventDefault();
    console.log(e);

    const regDetails = {
      name: name,
      email: email,
      password: password,
      city: city,
      age: age,
    };
    console.log("56", regDetails);
    if (name === "") {
      setError("Username feild is required");
    } else if (!email.includes("@")) {
      setError("email feild is required");
    } else if (password === "") {
      setError("password feild is required");
    } else if (password.length < 5) {
      setError("password length greater 5");
    } else if (city === "") {
      setError("city feild is required");
    } else if (age === "") {
      setError("age feild is required");
    } else {
      setError("");
      localStorage.setItem("data", JSON.stringify(regDetails));
      navigate("/login");
    }
  };

  return (
    <div>
      <div>
        <h1>User Registration</h1>
      </div>
      {error.length > 0 && <p data-testid="error">{error}</p>}
      <form>
        <label className="label">UserName</label>
        <input
          className="input"
          type="text"
          value={name}
          onChange={(e) => setName(e.target.value)}
          data-testid="username"
        />
        <label className="label">Email</label>
        <input
          className="input"
          type="text"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
          data-testid="email"
        />
        <label className="label">Password</label>
        <input
          className="input"
          type="password"
          value={password}
          onChange={(e) => setPassword(e.target.value)}
          data-testid="password"
        />
        <label className="label">City</label>
        <input
          className="input"
          type="city"
          value={city}
          onChange={(e) => setCity(e.target.value)}
          data-testid="city"
        />
        <label className="label">Age</label>
        <input
          className="input"
          type="age"
          value={age}
          onChange={(e) => setAge(e.target.value)}
          data-testid="age"
        />
        <button className="btn" onClick={(e) => handleAdd(e)}>
          submit
        </button>
      </form>
    </div>
  );
}
