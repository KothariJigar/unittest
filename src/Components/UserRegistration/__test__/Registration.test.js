import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

import Registration from "../Registration";

beforeEach(() => {
  render(
    <BrowserRouter>
      <Registration />
    </BrowserRouter>
  );
});

describe("Registration component tests", () => {
  it("should render login page with two input fields and a button", () => {
    const username = screen.getByTestId("username");
    const email = screen.getByTestId("email");
    const password = screen.getByTestId("password");
    const city = screen.getByTestId("city");
    const age = screen.getByTestId("age");
    const registerButton = screen.getByRole("button");

    expect(username).toBeInTheDocument();
    expect(email).toBeInTheDocument();
    expect(password).toBeInTheDocument();
    expect(city).toBeInTheDocument();
    expect(age).toBeInTheDocument();
    expect(registerButton).toBeInTheDocument();
  });

  it("should show username value on change", () => {
    const username = screen.getByTestId("username");
    fireEvent.change(username, { target: { value: "test" } });
    expect(username).toHaveValue("test");
  });

  it("should show email value on change", () => {
    const email = screen.getByTestId("email");
    fireEvent.change(email, { target: { value: "email" } });
    expect(email).toHaveValue("email");
  });

  it("should show password value on change", () => {
    const password = screen.getByTestId("password");
    fireEvent.change(password, { target: { value: "passwordtest" } });
    expect(password).toHaveValue("passwordtest");
  });

  it("should show city value on change", () => {
    const city = screen.getByTestId("city");
    fireEvent.change(city, { target: { value: "city" } });
    expect(city).toHaveValue("city");
  });

  it("should show age value on change", () => {
    const age = screen.getByTestId("age");
    fireEvent.change(age, { target: { value: "age" } });
    expect(age).toHaveValue("age");
  });

  it("should handle register click", () => {
    const registerButton = screen.getByRole("button");
    fireEvent.click(registerButton);
    const error = screen.getByTestId("error");
    expect(error).toBeInTheDocument();

    const username = screen.getByTestId("username");
    fireEvent.change(username, { target: { value: "test" } });
    fireEvent.click(registerButton);
    expect(error).toBeInTheDocument();

    const email = screen.getByTestId("email");
    fireEvent.change(email, { target: { value: "email@test.com" } });
    fireEvent.click(registerButton);
    expect(error).toBeInTheDocument();

    fireEvent.change(screen.getByTestId("password"), {
      target: { value: "p" },
    });
    fireEvent.click(registerButton);
    expect(error).toBeInTheDocument();

    const password = screen.getByTestId("password");
    fireEvent.change(password, { target: { value: "passwordtest" } });
    fireEvent.click(registerButton);
    expect(error).toBeInTheDocument();

    const city = screen.getByTestId("city");
    fireEvent.change(city, { target: { value: "city" } });
    fireEvent.click(registerButton);
    expect(error).toBeInTheDocument();

    const age = screen.getByTestId("age");
    fireEvent.change(age, { target: { value: "age" } });
    fireEvent.click(registerButton);
    const errorRemoved = screen.queryByTestId("error");
    expect(errorRemoved).not.toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(
      <BrowserRouter>
        <Registration />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
afterEach(cleanup);
