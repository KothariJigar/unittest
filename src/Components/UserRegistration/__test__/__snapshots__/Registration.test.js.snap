// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration component tests should match snapshot 1`] = `
<div>
  <div>
    <div>
      <h1>
        User Registration
      </h1>
    </div>
    <form>
      <label
        class="label"
      >
        UserName
      </label>
      <input
        class="input"
        data-testid="username"
        type="text"
        value=""
      />
      <label
        class="label"
      >
        Email
      </label>
      <input
        class="input"
        data-testid="email"
        type="text"
        value=""
      />
      <label
        class="label"
      >
        Password
      </label>
      <input
        class="input"
        data-testid="password"
        type="password"
        value=""
      />
      <label
        class="label"
      >
        City
      </label>
      <input
        class="input"
        data-testid="city"
        type="city"
        value=""
      />
      <label
        class="label"
      >
        Age
      </label>
      <input
        class="input"
        data-testid="age"
        type="age"
        value=""
      />
      <button
        class="btn"
      >
        submit
      </button>
    </form>
  </div>
</div>
`;
